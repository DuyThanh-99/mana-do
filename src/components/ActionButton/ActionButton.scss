.Action__btn {
  outline: none;
  border: none;
  box-shadow: 2px 0 2px currentColor;
  border-radius: 4px;
  min-height: 32px;
  min-width: 80px;
  padding: 4px 8px;

  &--checked {
    background-color: #a5def2;
  }
  &:hover {
    opacity: 0.6;
  }
}
