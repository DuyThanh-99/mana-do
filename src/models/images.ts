import ManabieLogo from '../assets/images/m-logo.jpg';

export { ManabieLogo };