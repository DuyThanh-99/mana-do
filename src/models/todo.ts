export enum TodoStatus {
  ACTIVE = 'ACTIVE',
  COMPLETED = 'COMPLETED'
}

export interface Todo {
  id: string
  user_id: string
  content : string
  status?: TodoStatus
  created_date: string
}

//I just need id and content passed to Edit Todo
export type TodoEdited  = Pick<Todo, "id" | "content">;