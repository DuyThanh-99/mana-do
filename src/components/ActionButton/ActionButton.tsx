import React from "react";
import "./ActionButton.scss";

export const ActionButton: React.FC<IActionButton> = ({
  actionBtn,
  statusCheck,
  border,
  boxShadow,
}) => {
  const { title, status, onClick } = actionBtn;
  return (
    <button
      className={`Action__btn ${
        status && status === statusCheck ? "Action__btn--checked" : ""
      }`}
      onClick={onClick}
      style={{ border, boxShadow }}
    >
      {title}
    </button>
  );
};
