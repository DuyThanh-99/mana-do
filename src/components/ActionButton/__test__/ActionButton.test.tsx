import React from "react";
import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import { ActionButton } from "../ActionButton";
import { TodoStatus } from "../../../models/todo";

const onAction = jest.fn();

//IBtn object
const actionButton: IBtn = { title: "All", status: "ALL", onClick: onAction };
const clearButton: IBtn = {
  title: "Clear all todos",
  status: "None",
  onClick: onAction,
};
// render
const renderComponent = (props: IActionButton) => {
  const { actionBtn, statusCheck } = props;
  return render(
    <ActionButton actionBtn={actionBtn} statusCheck={statusCheck} />
  );
};

// test case
describe("<ActionButton/>", () => {
  it("should match snapshot", () => {
    const wrapper = renderComponent({
      actionBtn: actionButton,
      statusCheck: "All",
    });
    expect(wrapper.container).toMatchSnapshot();
  });

  it("should match snapshot-clearButton", () => {
    const wrapper = renderComponent({
      actionBtn: clearButton,
    });
    expect(wrapper.container).toMatchSnapshot();
  });

  it("should render UI correctly", () => {
    renderComponent({
      actionBtn: actionButton,
      statusCheck: "ALL",
    });
    const title = screen.getByText("All");
    expect(title).toBeInTheDocument();
  });

  it("should be clickable correctly", () => {
    renderComponent({
      actionBtn: actionButton,
      statusCheck: "ALL",
    });
    const btn = screen.getByRole("button");
    expect(btn).not.toBeDisabled();
  });

  it("should be have current color on button when click correctly status", () => {
    renderComponent({
      actionBtn: actionButton,
      statusCheck: "ALL",
    });
    const btn = screen.getByRole("button");
    expect(btn).toHaveClass("Action__btn--checked");
  });
});
