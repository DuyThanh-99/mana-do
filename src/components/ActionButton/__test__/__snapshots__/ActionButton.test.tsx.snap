// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ActionButton/> should be render more button in Array 1`] = `
<div>
  <button
    class="Action__btn "
  >
    All
  </button>
  <button
    class="Action__btn Action__btn--checked"
  >
    Active
  </button>
  <button
    class="Action__btn "
  >
    Completed
  </button>
</div>
`;

exports[`<ActionButton/> should match snapshot 1`] = `
<div>
  <button
    class="Action__btn "
  >
    All
  </button>
</div>
`;

exports[`<ActionButton/> should match snapshot-clearButton 1`] = `
<div>
  <button
    class="Action__btn "
  >
    Clear all todos
  </button>
</div>
`;
